app/models/notification.rb

```diff
@@ -0,0 +1,10 @@
+class Notification < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :source, polymorphic: true
+
+  class << self
+    def unread_for_user(id)
```

Comment:
You can say its just a duck, you don't depend on the user model per say, but something that responds to `#id`. You can see Rails itself using that on numerous of occasions.